zeroex/order.go

```diff
@@ -154,6 +171,25 @@ func (s *SignedOrder) ComputeOrderHash() (common.Hash, error) {
 	return hash, nil
 }
 
+// ECSign signs the order with it's `makerAddress` and formats it to be a valid 0x order signature
+func (s *SignedOrder) ECSign(rpcClient *rpc.Client) ([]byte, error) {
+	orderHash, err := s.ComputeOrderHash()
+	if err != nil {
+		return nil, err
+	}
+	ecSignature, err := ethereum.ECSign(orderHash.Bytes(), s.MakerAddress, rpcClient)
+	if err != nil {
+		return nil, err
+	}
+
+	signature := []byte{}
+	signature = append(signature, ecSignature.V)
```

Comment:
Using an array instead of a slice will also make this code more efficient. Instead of making 4 calls to `append`, we can preallocate the length we need (66 bytes I think?). Something like this:

```go
signature := make([]byte, 66)
signature[0] = ecSignature.V
signature[1:33] = ecSignature.R
signature[34:66] = ecSignature.S
signature[66] = byte(EthSignSignature)
```